src/status_im/chat/models/mentions.cljs

```diff
@@ -0,0 +1,121 @@
+        lcase-alias (string/lower-case alias)
+        regex       (re-pattern
+                     (string/join
+                      "|"
+                      [(str "^" lcase-name "\\s")
+                       (str "^" lcase-name "$")
+                       (str "^" lcase-alias "\\s")
+                       (str "^" lcase-alias "$")]))
+        lcase-text  (string/lower-case text)]
+    (re-find regex lcase-text)))
+
+(defn match-mention
+  ([text users mention-key-idx]
+   (match-mention text users mention-key-idx (inc mention-key-idx) []))
+  ([text users mention-key-idx next-word-idx words]
+   (let [trimmed-text (subs text next-word-idx)]
+     (when-let [word (string/trim (re-find word-regex trimmed-text))]
+       (let [words           (conj words word)
+             searched-text   (string/lower-case (string/join " " words))
+             suggestions     (filter
+                              (fn [[_ {:keys [alias name]}]]
+                                (let [names (set [alias name])]
+                                  (some
+                                   (fn [username]
+                                     (string/starts-with?
+                                      (string/lower-case username)
+                                      searched-text))
+                                   names)))
+                              users)
+             suggestions-cnt (count suggestions)]
+         (cond (zero? suggestions-cnt)
+               nil
+
+               (and (= 1 suggestions-cnt)
+                    (mentioned? (second (first suggestions))
+                                (subs text (inc mention-key-idx))))
+               (second (first suggestions))
+
+               (> suggestions-cnt 1)
+               (let [word-len        (count word)
+                     text-len        (count text)
+                     next-word-start (+ next-word-idx (inc word-len))]
+                 (when (> text-len next-word-start)
+                   (match-mention text users mention-key-idx
+                                  next-word-start words)))))))))
+
+(defn replace-mentions
+  ([text users]
+   (replace-mentions text users 0))
+  ([text users idx]
+   (let [mention-key-idx (string/index-of text "@" idx)]
+     (if-not mention-key-idx
+       text
+       (let [{:keys [public-key alias]} (match-mention text users mention-key-idx)]
+         (if-not alias
+           (recur text users (inc mention-key-idx))
+           (let [new-text (string/join
+                           [(subs text 0 (inc mention-key-idx))
+                            public-key
+                            (subs text (+ (inc mention-key-idx)
+                                          (count alias)))])
+                 mention-end (+ (inc mention-key-idx) (count public-key))]
+             (recur new-text users mention-end))))))))
+
+(defn check-mentions [cofx text]
+  (replace-mentions text (get-mentionable-users cofx)))
```

Comment:
There are two ways to know whether there are mentions in the message:
1. Count `@` entries on text changes 
2. Check for `@` after pressing send button

The first one might require a bit complicated logic because of possibility of copy/pasting/cutting and should be actually implemented only if we find out that the second one is not sufficient enough. Currently we check an entire message for `@` and if it is not found we stop checking.

Reply:
yeah but i meant `get-mentionable-users` it will be called anyway , if there are no mentions at all, so i suggest something like
```
(defn check-mentions [cofx text]
  (if (string/index-of text "@" 0)
     (replace-mentions text (get-mentionable-users cofx)))
     text)
```
or move `get-mentionable-users ` inside `replace-mentions` 

```
(if-not mention-key-idx
       text
       (let [users (get-mentionable-users cofx)
               {:keys [public-key alias]} (match-mention text users mention-key-idx) ]
```

Reply:
ok I see, I'll just wrap it into a function

Reply:
done